src/trade/fees/borrowing/index.ts

```diff
@@ -0,0 +1,199 @@
+      break;
+    }
+  }
+
+  return fee;
+};
+
+const getPairPendingAccFees = (
+  pairIndex: number,
+  currentBlock: number,
+  vaultTvl: number,
+  context: { pairs: BorrowingFee.Pair[]; openInterest: OpenInterest }
+): { accFeeLong: number; accFeeShort: number; delta: number } => {
+  const {
+    pairs,
+    openInterest: { long, short },
+  } = context;
+
+  const pair = pairs[pairIndex];
+
+  return getPendingAccFees(
+    pair.accFeeLong,
+    pair.accFeeShort,
+    long,
+    short,
+    pair.feePerBlock,
+    currentBlock,
+    pair.accLastUpdatedBlock,
+    vaultTvl
+  );
+};
+
+const getPairPendingAccFee = (
+  pairIndex: number,
+  currentBlock: number,
+  vaultTvl: number,
+  long: boolean,
+  context: { pairs: BorrowingFee.Pair[]; openInterest: OpenInterest }
+): number => {
+  const { accFeeLong, accFeeShort } = getPairPendingAccFees(
+    pairIndex,
+    currentBlock,
+    vaultTvl,
+    context
+  );
+  return long ? accFeeLong : accFeeShort;
+};
+
+const getGroupPendingAccFees = (
+  groupIndex: number,
+  currentBlock: number,
+  vaultTvl: number,
+  context: { groups: BorrowingFee.Group[] }
+): { accFeeLong: number; accFeeShort: number; delta: number } => {
+  const { groups } = context;
+  const group = groups[groupIndex];
+  return getPendingAccFees(
+    group.accFeeLong,
+    group.accFeeShort,
+    group.oiLong,
+    group.oiShort,
+    group.feePerBlock,
+    currentBlock,
+    group.accLastUpdatedBlock,
+    vaultTvl
+  );
+};
+
+const getGroupPendingAccFee = (
+  groupIndex: number,
+  currentBlock: number,
+  vaultTvl: number,
+  long: boolean,
+  context: { groups: BorrowingFee.Group[] }
+): number => {
+  const { accFeeLong, accFeeShort } = getGroupPendingAccFees(
+    groupIndex,
+    currentBlock,
+    vaultTvl,
+    context
+  );
+  return long ? accFeeLong : accFeeShort;
+};
+
+type GetPairGroupAccFeesDeltasContext = {
+  currentBlock: number;
+  vaultTvl: number;
+  groups: BorrowingFee.Group[];
+  pairs: BorrowingFee.Pair[];
+  pairOpenInterest: OpenInterest;
+};
+const getPairGroupAccFeesDeltas = (
+  i: number,
+  pairGroups: BorrowingFee.PairGroup[],
+  initialFees: BorrowingFee.InitialAccFees,
+  pairIndex: number,
+  long: boolean,
+  context: GetPairGroupAccFeesDeltasContext
+): { deltaGroup: number; deltaPair: number; beforeTradeOpen: boolean } => {
+  const group = pairGroups[i];
+  const beforeTradeOpen = group.block <= initialFees.block;
+
+  let deltaGroup, deltaPair;
+  if (i == pairGroups.length - 1) {
+    const { currentBlock, vaultTvl, groups, pairs, pairOpenInterest } = context;
+    deltaGroup = getGroupPendingAccFee(
+      group.groupIndex,
+      currentBlock,
+      vaultTvl,
+      long,
+      { groups }
+    );
+    deltaPair = getPairPendingAccFee(pairIndex, currentBlock, vaultTvl, long, {
+      pairs,
+      openInterest: pairOpenInterest,
+    });
+  } else {
+    const nextGroup = pairGroups[i + 1];
+    if (beforeTradeOpen && nextGroup.block <= initialFees.block) {
```

Comment:
contract changed to `<` instead of `<=`.
`if (beforeTradeOpen && nextGroup.block < initialFees.block)`